src/com/unbounce/encors/core.clj

```diff
@@ -171,5 +171,34 @@
         :else ;; perform cors validation
         (apply-cors-policy {:req req
                             :app app
+                            :apply-headers apply-headers
                             :origin origin
                             :cors-policy cors-policy})))))
+
+(defn wrap-cors
+  "Applies CORS middleware. "
+  [handler cors-policy]
+  (build-cors-wrapper handler cors-policy apply-ring-headers))
+
+;; -- Aleph support  -----------------------------------------------------------
+;; Taken from aleph.http.middleware to optionally support aleph
+(def manifold-enabled?
+  (try
+    (require 'manifold.deferred)
+    true
+    (catch Throwable _ false)))
+
+(defn wrap-cors-aleph
+  "Aleph-compatible CORS middleware.
+
+  You should only use this if you have alelph/manifold available on the
+  classpath."
+  {:pre [manifold-enabled?]}
+  [handler cors-policy]
+  (let [deferred? (ns-resolve 'manifold.deferred 'deferred?)
+        chain'    (ns-resolve 'manifold.deferred 'chain')]
+    (build-cors-wrapper handler cors-policy
+                        (fn -apply-deferred-headers [response headers]
+                          (if (deferred? response)
+                            (chain' response #(apply-ring-headers % headers))
+                            (apply-ring-headers response headers))))))
```

Comment:
I would consider moving `wrap-cors-aleph` to `aleph.wrap-cors` (i.e. a different ns), so `manifold-enabled?` could be removed and `manifold.deferred` could be regularly required. It would be simpler and cleaner IMO.